libs/blocks/global-navigation/utilities/getUserEntitlements.js

```diff
@@ -63,27 +130,37 @@ const getSubscriptions = async ({ queryParams, locale }) => {
  * @param {object} object required params
  * @param {array} object.params array of name value query parameters [{name: 'Q', value: 'PARAM'}]
  * @param {object} object.locale {country: 'CH', language: 'de'}
+ * @param {string} object.format format function, raw or default
  * @returns {object} JIL Entitlements
  */
-const getUserEntitlements = async ({ params, locale } = {}) => {
-  if (!window.adobeIMS?.isSignedInUser()) return Promise.reject(new Error('User not signed in'));
+const getUserEntitlements = async ({ params, locale, format } = {}) => {
+  if (!window.adobeIMS?.isSignedInUser()) return Promise.resolve(emptyEntitlements());
 
   const queryParams = getQueryParameters(params);
-  if (entitlements[queryParams]) return entitlements[queryParams];
+  if (entitlements[queryParams]) {
+    return format === 'raw'
+      ? entitlements[queryParams]
+      : entitlements[queryParams]
+        .then((res) => mapSubscriptionCodes(res));
+  }
 
   let resolve;
```

Comment:
looks like concurrent calls might override resolve object.
Why not move `setTimeout` logic inside the promise and remove `resolve` variable?

Reply:
Moved a bit into the promise itself and added an abort controller for the setTimeout, feel like that's a little cleaner